spring-cloud-sleuth-instrumentation/src/main/java/org/springframework/cloud/sleuth/instrument/jdbc/TraceListenerStrategy.java

```diff
@@ -429,7 +434,15 @@ void afterConnectionClose(CON connectionKey, @Nullable Throwable t) {
 	}
 
 	private String spanName(String sql) {
-		return sql.substring(0, sql.indexOf(' ')).toLowerCase(Locale.ROOT);
+		String lowercaseSql = sql.toLowerCase(Locale.ROOT);
+		String spanName = DEFAULT_SPAN_NAME;
+		for (String spanNameCandidate : KNOWN_SPAN_NAMES) {
+			if (lowercaseSql.startsWith(spanNameCandidate)) {
+				spanName = spanNameCandidate;
+				break;
+			}
+		}
+		return spanName;
```

Comment:
Maybe a bit premature optimization, but what do you think about
```
String prefix = sql.substing(0, Math.min(6, sql.length()).toLowerCase(Locale.ROOT);
return KNOWN_SPAN_NAMES.contains(prefix) ? prefix : DEFAULT_SPAN_NAME
```
so that we generate less garbage by not lower-casing whole query. `KNOWN_SPAN_NAMES` can remain a list, that's likely faster than hashing anyway.